icaruscode/Filters/FilterCRTPMTMatching_module.cc

```diff
@@ -13,556 +13,607 @@
-            fOpHitA.clear();
-            fOpFlashPos.clear();
-            fMatchedCRTmodID.clear();
-            fMatchedCRTpos.clear();
-            fMatchedCRTtime.clear();
-            fMatchedCRTregion.clear();
-            fMatchedCRTsys.clear();
-            fMatchedCRTamplitude.clear();
-            fTofOpFlash.clear();
-            fCRTGateDiff.clear();
-        } // for Flash
+        for (auto& exiting : CRTmatches.exiting) {
+          vector<double> CRTpos = {exiting.CRTHit->x_pos, exiting.CRTHit->y_pos,
+                                   exiting.CRTHit->z_pos};
+          geo::Point_t thisCRTpos = {exiting.CRTHit->x_pos,
+                                     exiting.CRTHit->y_pos,
+                                     exiting.CRTHit->z_pos};
+          double CRTtime = exiting.CRTHit->ts1_ns / 1e3;
+          int CRTRegion = exiting.CRTHit->plane;
+          int CRTSys = 0;
+          if (CRTRegion >= 36) CRTSys = 1;
+          double CRTTof_opflash = CRTtime - tflash;
+          std::vector<int> HitsFebs;
+          for (auto& crts : exiting.CRTHit->feb_id) {
+            HitsFebs.emplace_back((int)crts);
+          }
+          if (CRTSys == 0) TopEx++;
+          if (CRTSys == 1) SideEx++;
+          MatchedCRT thisCRTMatch = {.CRTHitPos = thisCRTpos,
+                                     .CRTPMTTimeDiff_ns = CRTTof_opflash,
+                                     .CRTTime_us = CRTtime,
+                                     .CRTSys = CRTSys,
+                                     .CRTRegion = CRTRegion};
+          thisFlashCRTmatches.push_back(thisCRTMatch);
+          HitsFebs.clear();
+        }
+      }
+      FlashType thisFlashType = {.FlashPos = flash_pos,
+                                 .FlashTime_us = tflash,
+                                 .FlashGateTime_ns = fThisRelGateTime,
+                                 .inBeam = fThisInTime_beam,
+                                 .inGate = inTime,
+                                 .Classification = eventType,
+                                 .CRTmatches = thisFlashCRTmatches};
+      if (inTime == true) thisEventFlashes.push_back(thisFlashType);
     }
-    fEventType   = Type;
-
-    if (fFilterLevel == "loose") Cosmic = false;  // By default, with loose Filtering, everything is "intersting", nothing tagged as clear cosmic
-    else if (fFilterLevel == "medium") {
-	if (fEventType == 1 || fEventType == 3 || fEventType == 6 || fEventType == 7) Cosmic = true;
-	// With Medium filter, everything (inTime) which is associated with Top CRT Hit before the Flash is filtered as clear cosmic
-	else Cosmic = false;
+  }
+  bool HitsInGate = false;
+  if (fFilterLevel == "loose")
+    HitsInGate = false;  // By default, with loose Filtering, everything is
+                         // "intersting", nothing tagged as clear cosmic
+  else if (fFilterLevel == "medium") {
+    HitsInGate = true;
+    for (const auto& h : thisEventFlashes) {
+      bool ThisFlashType = false;
```

Comment:
More appropriate variable name? For example:
```suggestion
      bool isCosmic = false;
```